src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -179,7 +179,7 @@ public Task<SyntaxTree> GetSyntaxTreeAsync(CancellationToken cancellationToken =
             }
 
             // do it async for real.
-            return DocumentState.GetSyntaxTreeAsync(cancellationToken);
+            return DocumentState.GetSyntaxTreeAsync(cancellationToken).AsTask();
```

Comment:
You should probably update src/NuGet/Microsoft.CodeAnalysis.Workspaces.Common.nuspec to state the dependency too. It's "fine" for now in that workspaces depends on common, and it depends on extensions, but if ever somebody "fixes" the core one to not depend on it, this is now broken.